operators/swap_strips.py

```diff
@@ -153,3 +182,31 @@ def reconstruct(self, strip, init_channel, group, context):
                 if u.channel == channel and u != s:
                     u.channel += 1
             s.channel = channel
+            
+    def closest_direction_strip(self, context, strip):
+        """
+        Finds the closest strip to a given strip in a specific direction.
+        Args:
+        - strip: The base strip.
+        Returns: The closest strip to the given strip, in the proper direction.
+                 If no strip is found, returns None.
+        """
+        view2d = bpy.context.region.view2d
+        
+        if "up" == self.direction:
+            strips = [s for s in context.sequences if s.channel > strip.channel]
+        elif "down" == self.direction:
+            strips = [s for s in context.sequences if s.channel < strip.channel]
+        elif "left" == self.direction:
+            strips = [s for s in context.sequences if s.frame_final_end < \
+                    strip.frame_final_start]
+        elif "right" == self.direction:
+            strips = [s for s in context.sequences if s.frame_final_start > \
+                    strip.frame_final_end]
+                    
+        if not strips:
+            return None
+            
+        start_x, start_y = view2d.view_to_region(strip.frame_final_start, \
+                    strip.channel + 0.5)
+        return find_closest_strip(strips, start_x, start_y)

```

Comment:
This should work for left and right, but not up and down. Up and down is tricky actually, I'm not sure how to do this out of the top of my head